packages/devtools_app/lib/src/performance/flutter/performance_screen.dart

```diff
@@ -39,6 +41,31 @@ class PerformanceScreen extends Screen {
   @override
   String get docPageId => 'performance';
 
+  @visibleForTesting
+  static const debugModeWarning =
+      DebugModePerformanceMessage(DevToolsScreenType.performance);
+  @visibleForTesting
+  static const profileGranularityWarning =
+      HighProfileGranularityMessage(DevToolsScreenType.performance);
+
+  @override
+  List<Widget> messages(BuildContext context) {
+    final showDebugModeWarning = serviceManager.connectedApp.isFlutterAppNow &&
+        !serviceManager.connectedApp.isProfileBuildNow;
+    final performanceController = Controllers.of(context).performance;
+    final showProfileGranularityWarning = performanceController
+            .cpuProfilerController
+            .service
+            .profileGranularityFlagNotifier
```

Comment:
it would be cleaner to listen for the value of profileGranularityFlagNotifier
instead of using the current value.

Reply:
done